rero_ils/modules/patrons/listener.py

```diff
@@ -38,3 +42,30 @@ def enrich_patron_data(sender, json=None, record=None, index=None,
             json['organisation'] = {
                 'pid': org_pid
             }
+
+
+def create_subscription_patron_transaction(sender, record=None, **kwargs):
+    """This method check the patron to know if a subscription is requested.
+
+    This method checks the patron_type linked to a patron and determine if
+    a new subscription patron transaction should be created or not. If a
+    subscriotion is needed, then the 'subscription' patron attribute will be
+    updated.
+    This method should be connect with 'before_record_insert' and
+    'before_record_update'.
+
+    :param record: the record being performed
+    """
+    if record.get('$schema') != get_schema_for_resource(Patron):
+        return
```

Comment:
Should we return or raise an error?

Reply:
This is to filter the record to process. Each time a record is create/update, the signal is emit and the listener is called. But this listener must only be performed if it's a `Patron` record. If the record isn't a `Patron` then do nothing and return (not raise an error)